### ISSUE TITLE
aggregated proof verification in Solidity

### ISSUE THREAD

**Issue by jasonmorton** [zkonduit/ezkl#117]

**Comment by ethan-crypto** [comment]
I am gonna take a crack at optimizing the fix_verifier_sol.

**Comment by ethan-crypto** [comment]
I fixed the issue here: https://github.com/zkonduit/ezkl/pull/246

For all of the test cases in [TESTS_SOLIDITY](https://github.com/zkonduit/ezkl/blob/main/tests/integration_tests.rs#LL308C1-L308C1 run the ) I was able to generate and deploy aggr solidi verifiers. The fix was relatively simple: just adjust the Solc optimizer settings by enabling the optimizer and adjusting the optimzer-runs setting to the lowest possible value to optimize for deployment cost over execution cost, as there are trade offs between the two optimization routes. 

I don't think this optimization-runs = 1 should be hard coded and applied to every verifier, since not every verifier needs this optimization to be deployed. So rather than hard coding it I propose that we add it as optional arguments, say `optimizer: bool,  optimization_runs: usize`, to the CreateEVMVerifier and CreateEVMVerifierAggr commands. We can set the default values for the CreateEVMVerifier to be `true, 1` and then CreateEVMVerifierAggr `false, 200`. ( I set it to 200 as thats the default value for the optimizer-runs param in the official solc implementation).
